### DISCUSSION TITLE
Use replacement with sbt manager

### DISCUSSION THREAD

**Discussion by gaeljw** [renovatebot/renovate#25637] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Given the following repository: https://github.com/gaeljw/renovate-replacement-play, I'd like to get a PR opened using the "replacement" feature because one of my dependencies got its name changed.

- `"com.typesafe.play" % "sbt-plugin"` is renamed to `"org.playframework" % "sbt-plugin"`

### What you tried so far.

I've added the following replacement rules to my `renovate.json` config:

```json
"packageRules": [
      {
        "matchPackageNames": [
          "com.typesafe.play:sbt-plugin"
        ],
        "replacementName": "org.playframework:sbt-plugin",
        "replacementVersion": "3.0.0"
      }
    ]
```

But so far, I don't get any replacement PR opened. I only get new versions on the "old" naming, I'd like to get additional PRs with the new name and version 3.0.0.

### Relevant debug logs

<details><summary>Logs</summary>

```
"sbt": [
      {
        "deps": [
          {
            "datasource": "sbt-plugin",
            "depName": "com.typesafe.play:sbt-plugin",
            "packageName": "com.typesafe.play:sbt-plugin",
            "currentValue": "2.9.0",
            "depType": "plugin",
            "registryUrls": [
              "https://repo.maven.apache.org/maven2",
              "https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases"
            ],
            "updates": [
              {
                "updateType": "replacement",
                "newName": "org.playframework:sbt-plugin",
                "newValue": "3.0.0",
                "branchName": "renovate/com.typesafe.play-sbt-plugin-replacement"
              }
            ],
            "versioning": "ivy",
            "warnings": [],
            "sourceUrl": "https://github.com/playframework/playframework",
            "registryUrl": "https://repo.maven.apache.org/maven2",
            "dependencyUrl": "https://repo.maven.apache.org/maven2/com/typesafe/play/sbt-plugin",
            "currentVersion": "2.9.0",
            "fixedVersion": "2.9.0"
          }
        ],
        "packageFile": "project/plugins.sbt"
      }
    ]
  }

...

INFO: Cannot find replaceString in current file content. Was it already updated? (branch="renovate/com.typesafe.play-sbt-plugin-replacement")
{
  "packageFile": "project/plugins.sbt"
  "depName": "com.typesafe.play:sbt-plugin"
  "existingContent": "\naddSbtPlugin(\"com.typesafe.play\" % \"sbt-plugin\" % \"2.9.0\")\n"
  "replaceString": "2.9.0"
}

DEBUG: No content changed (branch="renovate/com.typesafe.play-sbt-plugin-replacement")
{
  "packageFile": "project/plugins.sbt"
  "depName": "com.typesafe.play:sbt-plugin"
}
```

</details>

I guess the error `Cannot find replaceString in current file content` is relevant but I'm not sure how to fix it.

**Comment by rarkins** [comment]
Can you simplify your reproduction so that:
- only one package 
- no other updates pending ?

**Reply by gaeljw** [reply]
Sure, done :)

I updated the logs in original description.

**Reply by rarkins** [reply]
The reason it doesn't work right now is that the package name is "split" within the file: https://github.com/gaeljw/renovate-replacement-play/blob/0495a4e6825f06218f728ce2cd57f68721eb9c5e/project/plugins.sbt#L2

So therefore any search/replace fails.

This might require SBT-specific replace code to be possible.

I'll create a feature request but I think this would require you or someone from the community for it to be implemented.

**Reply by gaeljw** [reply]
Ah right, makes sense. This would be the same with Maven.

I can give it a look if someone is able to give some pointers

**Reply by rarkins** [reply]
Not same for maven unless they also split package names up

**Reply by gaeljw** [reply]
@rarkins In Maven, you'd have:

```xml
<dependency>
  <groupId>com.typesafe.play</groupId>
  <artifactId>sbt-plugin</artifactId>
  <version>2.9.0</version>
</dependency>
```

**Reply by rarkins** [reply]
Of course, you're right. Unfortunately maven is excluded here: https://docs.renovatebot.com/configuration-options/#replacementname

**Reply by gaeljw** [reply]
Aaah, I see. I thought I had seen somewhere that all managers were supported now.

**Comment by rarkins** [comment, marked as answer]
Feature request created: #25643
